internal/server/v1/propertyvalues/core.go

```diff
@@ -34,50 +35,55 @@ import (
 func Fetch(
 	ctx context.Context,
 	store *store.Store,
-	properties []string,
 	entities []string,
+	properties []string,
 	limit int,
 	token string,
 	direction string,
 ) (
-	map[string]map[string][]*pb.EntityInfo,
+	map[string]map[string]map[string][]*pb.EntityInfo,
 	*pb.PaginationInfo,
 	error,
 ) {
 	var err error
+	propType, err := getEntityPropType(ctx, store.BtGroup, entities, properties, direction)
+	if err != nil {
+		return nil, nil, err
+	}
 	// Empty cursor groups when no token is given.
 	var cursorGroups []*pb.CursorGroup
 	if token == "" {
-		cursorGroups = buildDefaultCursorGroups(
-			properties, entities, len(store.BtGroup.Tables()))
+		cursorGroups = buildDefaultCursorGroups(entities, properties, propType, len(store.BtGroup.Tables()))
 	} else {
 		pi, err := pagination.Decode(token)
 		if err != nil {
-			return nil, nil, status.Errorf(
-				codes.InvalidArgument, "invalid pagination token: %s", token)
+			return nil, nil, status.Errorf(codes.InvalidArgument, "invalid pagination token: %s", token)
 		}
 		cursorGroups = pi.CursorGroups
 	}
 	if limit == 0 || limit > defaultLimit {
```

Comment:
nit: To make it more error-prone:

if limit <=0 || limit > defaultLimit {

?

Reply:
Done